core/cont/inc/TCollection.h

```diff
@@ -261,6 +261,13 @@ class TIter {
    Bool_t             operator!=(const TIter &aIter) const {
       return !(*this == aIter);
    }
+   TIter &operator=(TIterator *iter)
+   {
+      if (fIterator)
+         delete fIterator;
```

Comment:
```suggestion
      SafeDelete(fIterator);
```
to mimic the order deletion in this class.

Reply:
Here `SafeDelete` not necessary - fIterator is overwritten already with the next line

Reply:
Fair enough. I mis-read! :)